encfs/DirNode.cpp

```diff
@@ -739,8 +739,15 @@ int DirNode::unlink(const char *plaintextName) {
 
   Lock _lock(mutex);
 
+  size_t maxopen = 0;
+#ifdef __CYGWIN__
+  // When deleting a file, Windows first opens it, deletes it, and closes it
+  // We then have inevitably the file opened one time 
+  maxopen = 1;
+#endif
+
   int res = 0;
-  if ((ctx != nullptr) && ctx->lookupNode(plaintextName)) {
+  if ((ctx != nullptr) && (ctx->numberOfNode(plaintextName) > maxopen)) {
```

Comment:
Windows does not allow deleting open files, so we can just skip the check.
In other words:
```
if ( !windows && (ctx != nullptr) && (ctx->numberOfNode(plaintextName) > maxopen)) {
```

Reply:
I asked myself the same thing yesterday, but this code is even older than the git history, and I am not completely sure. It may be related to how encfs shares backing file descriptors between multiple open FUSE file descriptors. I believe they are stored in a table indexed by path.